### PR TITLE
test: migrate footer gauge header and heading tests

### PR DESCRIPTION
**Pull request by PavelKirvelTextkernel**
Link [task](https://textkernel.atlassian.net/browse/ONEUI-384)

### CODE REVIEW

File: src/components/Gauge/__tests__/Gauge.spec.tsx [hunk 1]
```diff
@@ -1,34 +1,41 @@
...
-        const wrapper = shallow(
+        const view = render(
             <Gauge metric="Metric" isContentLoading isProgressLoading>
                 Some content
             </Gauge>
         );
-        expect(wrapper.find('ContentPlaceholder').exists()).toBeTruthy();
+
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('alert')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/Gauge/__tests__/Gauge.spec.tsx:39]: image

File: src/components/Header/__tests__/Header.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                <IconTextkernel />
+            </a>
+        );
+        const view = render(<Header logo={logo}>This is a placeholder for children</Header>);
+        const images = screen.getAllByRole('img');
+
+        expect(view.container).toMatchSnapshot();
+        expect(images).toHaveLength(1);
+        expect(images[0]).toHaveAttribute('class', 'IconBase__svg');
+        expect(screen.getAllByRole('link')).toHaveLength(1);
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/Header/__tests__/Header.spec.tsx:40]: :+1:

File: src/components/Header/__tests__/Header.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        const logo = (
+            <a href="/test">
+                <IconTextkernel />
+            </a>
+        );
+        const view = render(<Header logo={logo}>This is a placeholder for children</Header>);
+        const images = screen.getAllByRole('img');
+
+        expect(view.container).toMatchSnapshot();
+        expect(images).toHaveLength(1);
```
[the remark this answers is not in the document]
Comment (PavelKirvelTextkernel) [reply src/components/Header/__tests__/Header.spec.tsx:38]: :+1:

**Review by andrentk — APPROVED** [review]

**Pull request merged**